File: 2019/day_1/day1.js
const fs = require('fs');
const data = fs.readFileSync('./input_day1.txt').toString().split('\n').map(Number);
const testInput = [14, 1969, 100756];

// Part 1, with function expression
const fule_calculator1 = (mass) => Math.floor(mass / 3 - 2);
let part1_output = data.map((mass) => fule_calculator1(mass))
                       .reduce((acc, val) => acc + val);

console.log(part1_output);

// Part 2
function fule_calculator2(mass, sum=0){
    let fule = fule_calculator1(mass);
    if (fule > 0){
        sum += fule
        return fule_calculator2(fule, sum)
    };
    return sum
};
let part2_output = data.map((mass) => fule_calculator2(mass))
                       .reduce((acc, val) => acc + val);
console.log(part2_output)

File: 2019/Day 1/day1.py
with open('input_day1.txt') as file:
    modules_masses = [x.strip() for x in file.read().splitlines()]

# Part 1
print('Part 1:', sum([(int(mass) // 3 - 2) for mass in modules_masses]))

# Part 2
def calculate_fule(mass, _sum=0):
    fule = mass // 3 - 2 
    if fule > 0:
        _sum += fule
        return calculate_fule(fule, _sum)
    else:
        return _sum

print('Part 2:', sum([calculate_fule(int(mass)) for mass in modules_masses]))